src/Tribe/Event_Status/Compatibility/Filter_Bar/Events_Status_Filter.php

```diff
@@ -0,0 +1,224 @@
+	 *
+	 * @since   TBD
+	 *
+	 * @var string
+	 */
+	public $name = 'event_status';
+
+	/**
+	 * Status Labels.
+	 *
+	 * @since TBD
+	 *
+	 * @var Status_Labels
+	 */
+	protected $status_labels;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since   TBD
+	 *
+	 * @param Status_Labels $status_labels An instance of the statuses handler.
+	 */
+	public function __construct( Status_Labels $status_labels ) {
+		$this->status_labels = $status_labels;
+		$name = $this->status_labels->get_event_status_label();
+
+		parent::__construct( $name, $this->slug );
+	}
+
+	/**
+	 * Returns the admin form HTML.
+	 *
+	 * @since   TBD
+	 *
+	 * @return string
+	 */
+	public function get_admin_form() {
+		return $this->get_title_field();
+	}
+
+	/**
+	 * Get the name for the admin field.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $name The individual name for the individual control (ie radio button).
+	 *
+	 * @return string The admin field input name.
+	 */
+	protected function get_admin_field_name( $name ) {
+		return "tribe_filter_options[{$this->slug}][{$name}]";
+	}
+
+	/**
+	 * Returns the value supported by this filter.
+	 *
+	 * @since TBD
+	 *
+	 * @param array<string|mixed> An array of values.
+	 */
+	protected function get_values() {
+		$default_values = [
+			'canceled'  => [
+				'name'  => _x( 'Hide canceled events', 'Canceled label for filter bar to hide canceled events.', 'the-events-calendar' ),
+				'value' => self::CANCELED,
+			],
+			'postponed' => [
+				'name'  => _x( 'Hide postponed events', 'Postponed label for filter bar to hide postponed events.', 'the-events-calendar' ),
+				'value' => self::POSTPONED,
+			],
+		];
+
+		/**
+		 * Allow filtering of the event statuses values that show in Filter Bar.
+		 *
+		 * @since TBD
+		 *
+		 * @param array<string|string> $default_values An array of filter values.
+		 */
+		return (array) apply_filters( 'tec_event_status_filterbar_values', $default_values );
+	}
+
+	/**
+	 * Sets up our join clause for the query.
+	 *
+	 * @since TBD
+	 */
+	protected function setup_join_clause() {
+		/** @var \wpdb $wpdb */
+		global $wpdb;
+
+		$clause = "LEFT JOIN {$wpdb->postmeta} AS {$this->alias}
+			ON ( {$wpdb->posts}.ID = {$this->alias}.post_id
+			AND {$this->alias}.meta_key = %s )";
+
+		$this->joinClause = $wpdb->prepare( $clause, Event_Status_Meta::$key_status );
+	}
+
+	/**
+	 * Sets up our where clause for the query.
+	 *
+	 * @since TBD
+	 */
+	protected function setup_where_clause() {
+		/** @var \wpdb $wpdb */
+		global $wpdb;
+
+		$clauses = $hide_clauses = [];
+
+		// Standard clauses to include so all events except for the selected ones to hide will be included.
+		$clauses[] = " {$this->alias}.meta_value = '' ";
+		$clauses[] = " {$this->alias}.meta_value IS NULL ";
+
+		if ( is_array( $this->currentValue ) ) {
+			$event_status_ids = implode( ',', array_map( 'esc_sql', $this->currentValue ) );
+		} else {
+			$event_status_ids = "'" . esc_sql( (string) $this->currentValue ) . "'";
+		}
```

Comment:
The `array_map` approach is missing the quoting of the strings. As an aside, I think this whole block can be simplified a bit:

```suggestion
		if ( ! empty( $this->currentValue ) ) {
			$event_status_ids = implode( "','", array_map( 'esc_sql', (array) $this->currentValue ) );
		}
```